### ISSUE TITLE
[require-await] Don't trigger when throwing errors

### ISSUE THREAD

**Issue by brandon-leapyear** [eslint/eslint#13272]
**What rule do you want to change?**
`require-await`

**Does this change cause the rule to produce more or fewer warnings?**
Fewer warnings

**How will the change be implemented? (New option, new default behavior, etc.)?**
Ideally, new default behavior, but it could be implemented as an option

**Please provide some example code that this change will affect:**

The following functions are different. The first returns `Promise.reject()`, and the second throws an error.

```js
async function foo() {
  throw new Error()
}

function foo() {
  throw new Error()
}
```

**What does the rule currently do for this code?**
`require-await` currently triggers if a function is marked as `async` AND does not use `await`

**What will the rule do after it's changed?**
`require-await` should be triggered if a function is marked as `async` AND (does not use `await` OR does not `throw` errors)

**Are you willing to submit a pull request to implement this change?**
Sure

**Comment by anikethsaha** [comment]
This issue has been [documented in the docs](https://eslint.org/docs/rules/require-await#when-not-to-use-it)

It is recommended to turn off the rule or disable the line.

**Comment by brandon-leapyear** [comment]
Thanks!
